esp/platform/espprotocol.cpp

```diff
@@ -22,27 +22,42 @@
 #include "platform.h"
 #include "espprotocol.hpp"
 #include "espbinding.hpp"
+#include "jatomic.hpp"
 #ifdef _USE_OPENLDAP
 #include "ldapsecurity.ipp"
 #endif
 
+RelaxedAtomic<unsigned> gActiveRequests;
+
+static std::shared_ptr<hpccMetrics::CustomMetric<unsigned>> pActiveRequests;
```

Comment:
The disadvantage of using T instead of atomic<T> for the parameter to the custom type is that this becomes more complex:

`static std::shared_ptr<hpccMetrics::CustomMetric<unsigned>> pActiveRequests;
`

However, maybe this would be better using a static initializer and avoiding any use of MODULE_INIT:

`static auto pActiveRequests = hpccMetrics::createCustomMetricAndAddToReporter("activerequests", "Number of active requests", hpccMetrics::METRICS_GAUGE, gActiveRequests);
`
I can't immediately think of any disadvantages - the main one would be the danger of functions being called before the base libraries are initialised - e.g. DBGLOG